Added Crows and Bread Crumbs

Shnupbups opened the pull request:
Added Crows. They currently don't spawn or have animations and only have a couple of sounds, but other than that they're pretty good.

Also added Bread Crumbs, craftable from Bread. You can eat them or place them down to make a trail, Hansel and Gretel style. But much like the fairy tale, Crows can eat them when placed down, destroying your trail.

Shnupbups committed:
Added Crows and Bread Crumbs

Added Crows. They currently don't spawn or have animations and only have a couple of sounds, but other than that they're pretty good.

Also added Bread Crumbs, craftable from Bread. You can eat them or place them down to make a trail, Hansel and Gretel style. But much like the fairy tale, Crows can eat them when placed down, destroying your trail.

Shnupbups commented:
The plan is to have a Gingerbread House structure spawn rarely in the Spooky Dimension, with Bread Crumbs leading to it. The structure will have a Witch Spawner as well as loot (and maybe some rotten flesh in a Smoker...).

Draylar commented on src/main/java/com/fabriccommunity/spookytime/block/BreadCrumbsBlock.java:
```diff
@@ -0,0 +1,44 @@
+import net.minecraft.world.ViewableWorld;
+
+public class BreadCrumbsBlock extends Block {
+	public static final VoxelShape SHAPE = Block.createCuboidShape(0.0D, 0.0D, 0.0D, 16.0D, 1.0D, 16.0D);
```
Should/could this be private?

Draylar commented on src/main/java/com/fabriccommunity/spookytime/block/BreadCrumbsBlock.java:
```diff
@@ -0,0 +1,44 @@
+		return bpe == BlockPlacementEnvironment.LAND;
+	}
+	
+	public VoxelShape getOutlineShape(BlockState state, BlockView blockView, BlockPos pos, EntityContext context) {
```
@Override

Draylar commented on src/main/java/com/fabriccommunity/spookytime/block/BreadCrumbsBlock.java:
```diff
@@ -0,0 +1,44 @@
+		super(settings);
+	}
+	
+	public boolean canPlaceAtSide(BlockState state, BlockView blockView, BlockPos pos, BlockPlacementEnvironment bpe) {
```
@Override

Draylar commented on src/main/java/com/fabriccommunity/spookytime/block/BreadCrumbsBlock.java:
```diff
@@ -0,0 +1,44 @@
+		return SHAPE;
+	}
+	
+	public boolean canPlaceAt(BlockState state, ViewableWorld world, BlockPos pos) {
```
@Override

Draylar commented on src/main/java/com/fabriccommunity/spookytime/block/BreadCrumbsBlock.java:
```diff
@@ -0,0 +1,44 @@
+		return Block.isFaceFullSquare(world.getBlockState(down).getCollisionShape(world, down), Direction.UP);
+	}
+	
+	public BlockState getStateForNeighborUpdate(BlockState state, Direction direction, BlockState state2, IWorld world, BlockPos pos, BlockPos pos2) {
```
@Override

Draylar commented on src/main/java/com/fabriccommunity/spookytime/entity/CrowEntity.java:
```diff
@@ -0,0 +1,160 @@
+import javax.annotation.Nullable;
+import java.util.Random;
+
+public class CrowEntity extends AnimalEntity implements Bird {
```
ctrl + f -> "_1" -> replace -> profit

Draylar commented on src/main/java/com/fabriccommunity/spookytime/entity/CrowEntity.java:
```diff
@@ -0,0 +1,160 @@
+		return false;
+	}
+	
+	@Nullable
```
Remove Nullable annotation

Draylar commented on src/main/java/com/fabriccommunity/spookytime/entity/goal/EatBreadcrumbsGoal.java:
```diff
@@ -0,0 +1,147 @@
+import javax.annotation.Nullable;
+import java.util.Random;
+
+public class EatBreadcrumbsGoal extends MoveToTargetPosGoal {
```
@Override and remove Nullable

Draylar commented on src/main/java/com/fabriccommunity/spookytime/entity/goal/TemptBirdGoal.java:
```diff
@@ -0,0 +1,111 @@
+
+import java.util.EnumSet;
+
+public class TemptBirdGoal extends Goal {
```
@Override and remove Nullable

Draylar reviewed (changes requested):
Reviewing on a phone is hard.

:crab: style police have arrived :crab:

Juuxel commented on src/main/java/com/fabriccommunity/spookytime/client/model/CrowEntityModel.java:
```diff
@@ -0,0 +1,76 @@
+	private final Cuboid right_leg;
+	private final Cuboid tail;
+	private final Cuboid left_wing;
+	private final Cuboid right_wing;
```
`camelCase` instead of `snake_case`

Juuxel commented on src/main/java/com/fabriccommunity/spookytime/client/render/CrowEntityRenderer.java:
```diff
@@ -0,0 +1,25 @@
+import com.fabriccommunity.spookytime.entity.CrowEntity;
+import com.fabriccommunity.spookytime.entity.PumpcownEntity;
+
+@SuppressWarnings({"rawtypes", "unchecked"})
```
What do you need these suppressions for? I don't see any generic types here.

Juuxel reviewed (changes requested):
:crab: more style comments :crab:

Shnupbups committed:
Better Crows + style fixes

Better crow model, more compact texture, behaviour tweaks, legs are now animated but wings aren't because that's beyond my ability...
also fixed style issues

Shnupbups committed:
Merge remote-tracking branch 'upstream/master' into features

Shnupbups committed:
fixes, finalisation, and style

fixed some things that went wrong when resolving conflicts, made a lot of things final, did style stuff

Juuxel commented on src/main/java/com/fabriccommunity/spookytime/registry/SpookyBlocks.java:
```diff
@@ -25,50 +20,42 @@
-	public static Block DEADER_BUSH = register("deader_bush", new DeaderBushBlock(FabricBlockSettings.copy(Blocks.DEAD_BUSH).materialColor(MaterialColor.BROWN).build()), new Item.Settings().group(SpookyTime.GROUP));
-
-
+	public static final Block TINY_PUMPKIN = register("tiny_pumpkin", new TinyPumpkinBlock(FabricBlockSettings.of(Material.PUMPKIN).strength(1.0F, 1.0F).sounds(BlockSoundGroup.LANTERN).build()), new Item.Settings().group(SpookyTime.GROUP).food(new FoodComponent.Builder().hunger(2).saturationModifier(0.4f).build()));
+	public static final Block WITCHED_PUMPKIN = register("witched_pumpkin", new TinyPumpkinBlock(Block.Settings.copy(TINY_PUMPKIN)), block -> new WitchedPumpkinItem(block, new Item.Settings().group(SpookyTime.GROUP).food(new FoodComponent.Builder().statusEffect(new StatusEffectInstance(StatusEffects.BLINDNESS, 200), 1).hunger(1).saturationModifier(0.1f).alwaysEdible().build())));
+	public static final Block DECEASED_DIRT = register("deceased_dirt", new Block(FabricBlockSettings.copy(Blocks.DIRT).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block DECEASED_GRASS_BLOCK = register("deceased_grass_block", new DeceasedGrassBlock(FabricBlockSettings.copy(Blocks.GRASS_BLOCK).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_SAND = register("tainted_sand", new FallingBlock(FabricBlockSettings.copy(Blocks.SAND).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_GRAVEL = register("tainted_gravel", new FallingBlock(FabricBlockSettings.copy(Blocks.GRAVEL).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_GLASS = register("tainted_glass", new TranslucentGlassBlock(FabricBlockSettings.copy(Blocks.GLASS).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_GLASS_PANE = register("tainted_glass_pane", new TranslucentGlassPaneBlock(FabricBlockSettings.copy(Blocks.GLASS_PANE).build()));
+	public static final Block TAINTED_SANDSTONE = register("tainted_sandstone", new Block(FabricBlockSettings.copy(Blocks.SANDSTONE).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block SMOOTH_TAINTED_SANDSTONE = register("smooth_tainted_sandstone", new Block(FabricBlockSettings.copy(Blocks.SMOOTH_SANDSTONE).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block CUT_TAINTED_SANDSTONE = register("cut_tainted_sandstone", new Block(FabricBlockSettings.copy(Blocks.CUT_SANDSTONE).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block CHISELED_TAINTED_SANDSTONE = register("chiseled_tainted_sandstone", new Block(FabricBlockSettings.copy(Blocks.CHISELED_SANDSTONE).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_SANDSTONE_STAIRS = register("tainted_sandstone_stairs", new SpookyStairsBlock(TAINTED_SANDSTONE, FabricBlockSettings.copy(Blocks.SANDSTONE_STAIRS).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block SMOOTH_TAINTED_SANDSTONE_STAIRS = register("smooth_tainted_sandstone_stairs", new SpookyStairsBlock(SMOOTH_TAINTED_SANDSTONE, FabricBlockSettings.copy(Blocks.SMOOTH_SANDSTONE_STAIRS).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_SANDSTONE_SLAB = register("tainted_sandstone_slab", new SlabBlock(FabricBlockSettings.copy(Blocks.SANDSTONE_SLAB).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block SMOOTH_TAINTED_SANDSTONE_SLAB = register("smooth_tainted_sandstone_slab", new SlabBlock(FabricBlockSettings.copy(Blocks.SMOOTH_SANDSTONE_SLAB).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block CUT_TAINTED_SANDSTONE_SLAB = register("cut_tainted_sandstone_slab", new SlabBlock(FabricBlockSettings.copy(Blocks.CUT_SANDSTONE_SLAB).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block TAINTED_SANDSTONE_WALL = register("tainted_sandstone_wall", new WallBlock(FabricBlockSettings.copy(Blocks.SANDSTONE_WALL).materialColor(MaterialColor.PURPLE).build()));
+	public static final Block SOUL_GLASS = register("soul_glass", new TranslucentGlassBlock(FabricBlockSettings.copy(Blocks.GLASS).materialColor(MaterialColor.BROWN).build()));
+	public static final Block SOUL_GLASS_PANE = register("soul_glass_pane", new TranslucentGlassPaneBlock(FabricBlockSettings.copy(Blocks.GLASS_PANE).build()));
+	public static final Block SPOOKIUM_ORE = register("spookium_ore", new OreBlock(FabricBlockSettings.copy(Blocks.IRON_ORE).build()), new Item.Settings().group(SpookyTime.GROUP).rarity(Rarity.EPIC));
+	public static final Block SPOOKIUM_BLOCK = register("spookium_block", new Block(FabricBlockSettings.copy(Blocks.IRON_BLOCK).materialColor(MaterialColor.RED).build()), new Item.Settings().group(SpookyTime.GROUP).rarity(Rarity.EPIC));
+	public static final Block WITCH_WATER_BLOCK = register("witch_water", new WitchWaterBlock(SpookyFluids.WITCH_WATER, FabricBlockSettings.copy(Blocks.WATER).build()), (BlockItem) null);
+	public static final Block SPOOKY_TREASURE_CHEST = register("spooky_treasure_chest", new SpookyTreasureChestBlock(FabricBlockSettings.of(Material.METAL).build()), new Item.Settings().group(SpookyTime.GROUP));
+	public static final Block BLOOD_BLOCK = register("blood", new BloodBlock(SpookyFluids.BLOOD, FabricBlockSettings.copy(Blocks.LAVA).lightLevel(0).build()), (BlockItem) null);
+	public static final Block SPOOKY_CACTUS = register("spooky_cactus", new SpookyCactusBlock(FabricBlockSettings.copy(Blocks.CACTUS).materialColor(MaterialColor.BROWN).build()), new Item.Settings().group(SpookyTime.GROUP));
+	public static final Block DEADER_BUSH = register("deader_bush", new DeaderBushBlock(FabricBlockSettings.copy(Blocks.DEAD_BUSH).materialColor(MaterialColor.BROWN).build()), new Item.Settings().group(SpookyTime.GROUP));
```
:+1:
